### PR TITLE
 random search when suggested parameter appeared in the history

### PR DESCRIPTION
**Pull request by leelaylay**
To avoid the waste of computer resource, I suggest  random search when suggested parameter appeared in the history for TPE and Anneal.

### CODE REVIEW

File: src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py [hunk 1]
```diff
@@ -211,41 +211,22 @@ def update_search_space(self, search_space):
...
         """
         Returns a set of trial (hyper-)parameters, as a serializable object.
-        
+
         Parameters
         ----------
         parameter_id : int
-
-        Returns
-        -------
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: why remove it?

File: src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py [hunk 2]
```diff
@@ -326,3 +307,36 @@ def miscs_update_idxs_vals(self, miscs, idxs, vals,
...
+            new_trials = hp.rand.suggest(new_ids, rval.domain, trials, random_state)
+        else:
+            new_trials = algorithm(new_ids, rval.domain, trials, random_state)
+        rval.trials.refresh()
+        vals = new_trials[0]['misc']['vals']
+        parameter = dict()
+        for key in vals:
+            try:
+                parameter[key] = vals[key][0].item()
+            except Exception:
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: please try to catch specific Exception here.

**Comment by xuehui1991** [comment]
Please test for this situation.

File: src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py (hunk 1 above)
Comment (leelaylay) [reply src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: remove it by accident

File: src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py (hunk 2 above)
Comment (leelaylay) [reply src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: OK

File: src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py [hunk 3]
```diff
@@ -220,32 +220,17 @@ def generate_parameters(self, parameter_id):
...
-        parameter = dict()
-        for key in vals:
-            try:
-                parameter[key] = vals[key][0].item()
-            except Exception:
-                parameter[key] = None
-
-        # remove '_index' from json2parameter and save params-id
-        total_params = json2parameter(self.json, parameter)
+        total_params = self.get_suggestion(random_search=False)
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: please add comments here to tell why making this change, and that it is still possible to generate duplicate configurations.

Comment (leelaylay) [reply src/sdk/pynni/nni/hyperopt_tuner/hyperopt_tuner.py]: Ok

**Pull request merged**